machine/src/lib.rs

```diff
@@ -40,14 +60,21 @@ where
             iface.add_ipv4_route(Ipv4Range::global().into()).unwrap();
 
             let iface = async_io::Async::new(iface).unwrap();
-
             let (mut tx, mut rx) = plug.split();
-            let (mut reader, mut writer) = iface.split();
 
-            let reader_task = async move {
+            let ctrl_task = async {
+                while let Some(ctrl) = ctrl.next().await {
+                    match ctrl {
+                        IfaceCtrl::Up => iface.get_ref().put_up().unwrap(),
+                        IfaceCtrl::Down => iface.get_ref().put_down().unwrap(),
```

Comment:
Will potentials panics be swallowed or bring down the whole process>

Reply:
should bring down the whole process